wp/lib/bap_wp/src/compare.ml

```diff
@@ -42,6 +44,43 @@ let set_to_eqs (env1 : Env.t) (env2 : Env.t) (vars : Var.Set.t) : Constr.t list
         in (eq::eqs, env1, env2)
       )
 
+(* This hypothesis needs to be generated after visiting both of the subs because
```

Comment:
Maybe a quick note about what it's actually computing might be useful.

Reply:
Can do.